### PR TITLE
OrgId check for invitation

### LINKED ISSUE #739: When a user is added to an org, clear other invitations
users can have only one org, clear other invites and block orgId updates on users

### PR DESCRIPTION
**Pull request by Mussuranae**
Check if user invited has already an orgId when we want to invite him/her on our organization
Fix #739

### CODE REVIEW

File: libs/invitation/src/lib/+state/invitation.service.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export class InvitationService extends CollectionService<InvitationState> {
 
   /** Return true if there is already a pending invitation for a list of users */
   public async orgInvitationExists(userEmails: string[]): Promise<boolean> {
```
Comment (phpgeekfr) [inline libs/invitation/src/lib/+state/invitation.service.ts:60]: maybe also rename `orgInvitationExists`

File: apps/backend-functions/src/invitation.ts [hunk 2]
```diff
@@ -190,13 +190,31 @@ That would have exceeded the current limit which is ${MEETING_MAX_INVITATIONS_NU
...
     .where('status', 'in', invitationStatus.filter(s => s !== 'declined'))
     .get());
   const invitationQuery = await Promise.all(invitationPromises);
-  return invitationQuery.some(d => d.docs.length > 0);
+
+  const isInvitationExist = invitationQuery.some(d => d.docs.length > 0);
+
+  if (isUserOrgIdExist === true && isInvitationExist === true) return true;
+  if (isUserOrgIdExist === true || isInvitationExist === true) return true;
+  else return false;
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.ts:219]: Why so much complexity ?
And the `if` at the end does not make sense.

You can do 
```
export async function hasUserAnOrgOrIsAlreadyInvited(userEmails: string[]) {

  const userPromises = userEmails.map(email => db.collection('users')
  .where('email', '==', email)
  .get());
  const userQuery = await Promise.all(userPromises);
  const orgIdExists = [];

  const someUserAlreadyHaveOrg = userQuery.some(d =>d.docs && d.docs.some(u => !!u.data().orgId));

  if(someUserAlreadyHaveOrg) { return true };

  const invitationPromises = userEmails.map(email => db.collection('invitations')
    .where('type', '==', 'joinOrganization')
    .where('toUser.email', '==', email)
    .where('status', 'in', invitationStatus.filter(s => s !== 'declined'))
    .get());
  const invitationQuery = await Promise.all(invitationPromises);

  return invitationQuery.some(d => d.docs.length > 0);
}
```

**Review by phpgeekfr — CHANGES_REQUESTED** [review]

File: apps/backend-functions/src/invitation.ts (hunk 2 above)
Comment (Mussuranae) [reply apps/backend-functions/src/invitation.ts]: Thanks, it's modified

File: apps/backend-functions/src/invitation.ts [hunk 3]
```diff
@@ -190,13 +190,22 @@ That would have exceeded the current limit which is ${MEETING_MAX_INVITATIONS_NU
...
-export async function isInvitationToJoinOrgExist(userEmails: string[]) {
+export async function hasUserAnOrgOrIsAlreadyInvited(userEmails: string[]) {
+
+  const userPromises = userEmails.map(email => db.collection('users')
+  .where('email', '==', email)
+  .get());
+  const userQuery = await Promise.all(userPromises);
+
+  const hasUserAlreadyAnOrg = userQuery.some(d => d.docs && d.docs.some(u => !!u.data().orgId));
+  if(hasUserAlreadyAnOrg) return true;
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.ts:201]: bad indent

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**